lib/zclient.c

```diff
@@ -1604,37 +1630,31 @@ lm_get_label_chunk (struct zclient *zclient, u_char keep, uint32_t chunk_size,
     zlog_debug ("%s: Label chunk request (%d bytes) sent", __func__, ret);
 
   /* read response */
-  s = zclient->ibuf;
-  stream_reset (s);
-
-  ret = zclient_read_header (s, zclient->sock, &size, &marker, &version,
-                             &vrf_id, &cmd);
-  if (ret != 0 || cmd != ZEBRA_GET_LABEL_CHUNK) {
-      zlog_err ("%s: Invalid Get Label Chunk Message Reply Header", __func__);
-      return -1;
-  }
-  if (zclient_debug)
-    zlog_debug ("%s: Label chunk response (%d bytes) received", __func__, size);
+  if (zclient_read_sync_response (zclient, ZEBRA_GET_LABEL_CHUNK) != 0)
```

Comment:
I don't think it's appropriate to block waiting for an external response here; this will lock up the entire zebra process if the external entity isn't responding for any random reason. IMHO we need something asynchronous here, unfortunately (yes it's quite a bit more work and management overhead...)